src/core/infer_request.cc

```diff
@@ -284,14 +284,41 @@ InferenceRequest::CopyAsNull(const InferenceRequest& from)
     *new_input->MutableShape() = input.second.Shape();
     *new_input->MutableShapeWithBatchDim() = input.second.ShapeWithBatchDim();
 
+    std::shared_ptr<MutableMemory> new_data_str;
+
+    // Custom handling for null requests in TYPE_STRING. The null requests in
+    // type string have to follow a certain structure. We set the first 4 bytes
+    // in the Tensor to represent the number of total available bytes.
+    if (inference::DataType::TYPE_STRING == input.second.DType()) {
+      new_data_str = std::make_shared<AllocatedMemory>(
+          input.second.Data()->TotalByteSize(), mem_type, mem_id);
+      uint32_t* new_data_base =
+          reinterpret_cast<uint32_t*>(new_data_str->MutableBuffer());
```

Comment:
The `new_data_base` doesn't seem to be used.

Reply:
It is used in the next line.

Reply:
My bad, I misread that. I think given that you are just allocating buffer for "num_elements" of empty string, you just need to initialize the whole buffer to zeros.